substrate/pkg/utils/helm/chart.go

```diff
@@ -0,0 +1,64 @@
+package helm
+
+import (
+	"context"
+	"fmt"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"knative.dev/pkg/logging"
+)
+
+type Chart struct {
+	Namespace       string
+	Name            string
+	Repository      string
+	Version         string
+	CreateNamespace bool
+	Values          map[string]interface{}
+	KubeConfig      string
+}
+
+func Apply(ctx context.Context, chart *Chart) error {
+	// Get the chart from the repository
+	charts, err := new(getter.HTTPGetter).Get(fmt.Sprintf("%s/%s-%s.tgz", chart.Repository, chart.Name, chart.Version))
```

Comment:
Why don't we pass the complete URL instead, since `chart.Repository` also contains version for some charts like EBS - https://github.com/kubernetes-sigs/aws-ebs-csi-driver/releases/download/helm-chart-aws-ebs-csi-driver-2.6.3/aws-ebs-csi-driver-2.6.3.tgz

Reply:
I think this is the exception and not the rule. I think in most cases (e.g. not EBS), its a simple string version bump. I could see doing it both ways, but found this approach to be more idiomatic. Happy to revert if you feel strongly.